source/collimation.f90

```diff
@@ -2341,6 +2374,10 @@ subroutine collimate_do_collimator(stracki)
   if(cdb_cSliced(icoll) > 0) then ! Treatment of sliced collimators
     ! Now, loop over the number of slices and call k2coll_collimate each time.
     ! For each slice, the corresponding offset and angle are to be used.
+    if(cdb_isCrystal(icoll)) then
+      write(lerr,"(a)") "COLL> ERROR A crystal collimator cannot be sliced"
+      call prror
+    end if
```

Comment:
In terms of logic, it's a lot easier to do it here because all of the slices have been processed, the DB and old/new COLL block parsed, etc. We could check it a bit further up in this subroutine, but not before it.